### PR TITLE
Comprehensive Code Cleanup and PostgreSQL Config Enhancements

### PR DESCRIPTION
**Pull request by ThomasSanson**
This pull request presents a variety of improvements and adjustments focused on enhancing the readability, maintainability, and performance of the code. These changes span multiple files and primarily involve formatting changes

### CODE REVIEW

File: add_balancer.yml [hunk 1]
```diff
@@ -11,14 +11,14 @@
 
   pre_tasks:
     - name: Include OS-specific variables
-      include_vars: "vars/{{ ansible_os_family }}.yml"
+      include_vars: vars/{{ ansible_os_family }}.yml
```
Comment (vitabaks) [inline add_balancer.yml:14]: in most cases, it is recommended to use double quotes around expressions containing Ansible variables. They allow the Jinja2 template language to correctly interpret and process these variables. Without quotes, Ansible may misinterpret some values, especially if they contain special characters or spaces.

File: add_balancer.yml [hunk 2]
```diff
@@ -40,10 +40,10 @@
       tags: always
 
     # Stop, if no nodes found with new_node variable
-    - name: "Pre-Check error. No nodes found with new_node set to true"
+    - name: Pre-Check error. No nodes found with new_node set to true
       run_once: true # noqa run-once
       fail:
-        msg: "Please specify the new_node=true variable for the new balancer server to add it to the existing cluster."
+        msg: Please specify the new_node=true variable for the new balancer server to add it to the existing cluster.
```
Comment (vitabaks) [inline add_balancer.yml:46]: as a best practice, I would recommend using double quotes for greater clarity and predictability.

File: config_pgcluster.yml [hunk 3]
```diff
@@ -179,7 +179,8 @@
               + [{ 'option': 'shared_preload_libraries', 'value': new_value }] }}
           vars:
             # Find the last item in postgresql_parameters where the option is 'shared_preload_libraries'
-            shared_preload_libraries_item: "{{ postgresql_parameters | selectattr('option', 'equalto', 'shared_preload_libraries') | list | last | default({ 'value': '' }) }}"
+            shared_preload_libraries_item: "{{ postgresql_parameters | selectattr('option', 'equalto', 'shared_preload_libraries') | list | last | default({ 'value':
+              '' }) }}"
```
Comment (vitabaks) [inline config_pgcluster.yml:183]: I don't understand the need for this change because a lot of spaces have been added here

I would not recommend changing such complex conditions

File: add_balancer.yml [hunk 4]
```diff
@@ -157,9 +157,7 @@
...
 
     - role: confd
-      when: dcs_type == "etcd" and
-        with_haproxy_load_balancing|bool
+      when: dcs_type == "etcd" and with_haproxy_load_balancing|bool
 
     - role: keepalived
-      when: cluster_vip is defined and cluster_vip | length > 0 and
-        with_haproxy_load_balancing|bool
+      when: cluster_vip is defined and cluster_vip | length > 0 and with_haproxy_load_balancing|bool
```
Comment (vitabaks) [inline add_balancer.yml:163]: perhaps the best practice would be to switch to lists so as not to make the string too long for large conditions

when:
  - condition1
  - condition2

File: deploy_pgcluster.yml [hunk 5]
```diff
@@ -43,9 +43,11 @@
         system_info:
           OS: "{{ ansible_distribution | default('N/A') }} {{ ansible_distribution_version | default('N/A') }}"
           Kernel: "{{ ansible_kernel | default('N/A') }}"
-          CPU model: "{{ ansible_processor[2] | default('N/A') }}, count: {{ ansible_processor_count | default('N/A') }}, cores: {{ ansible_processor_cores | default('N/A') }}"
+          CPU model: "{{ ansible_processor[2] | default('N/A') }}, count: {{ ansible_processor_count | default('N/A') }}, cores: {{ ansible_processor_cores | default('N/A')
+            }}"
```
Comment (vitabaks) [inline deploy_pgcluster.yml:47]: I don't see the need for such changes

File: roles/ansible-role-firewall/tasks/main.yml [hunk 6]
```diff
@@ -20,7 +19,7 @@
     dest: /etc/firewall.bash
     owner: root
     group: root
-    mode: 0744
+    mode: "0744"
```
Comment (vitabaks) [inline roles/ansible-role-firewall/tasks/main.yml:22]: why are quotation marks added here?

File: roles/confd/tasks/main.yml [hunk 7]
```diff
@@ -100,14 +99,15 @@
...
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.master }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.master }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_sync }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_sync }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_async }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_async }}' }
+        - regexp: ^.*bind.*:{{ haproxy_listen_port.stats }}$
+          line: "    bind {{ hostvars[inventory_hostname].inventory_hostname }}:{{ haproxy_listen_port.stats }}"
+        - { regexp: "^.*bind.*:{{ haproxy_listen_port.master }}$", line: "    bind {{ cluster_vip }}:{{ haproxy_listen_port.master }}" }
+        - { regexp: "^.*bind.*:{{ haproxy_listen_port.replicas }}$", line: "    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas }}" }
```
Comment (vitabaks) [inline roles/confd/tasks/main.yml:105]: why not stick to one style?

```
- regexp
  line
```
or
```
- { regexp, line }
```

Comment (vitabaks) [inline roles/confd/tasks/main.yml:105]: why not stick to one style?

```
- regexp
  line
```
or
```
- { regexp, line }
```

File: roles/locales/tasks/main.yml [hunk 8]
```diff
@@ -62,35 +58,33 @@
...
         line: "{{ item.line }}"
         owner: root
         group: root
-        mode: '0644'
+        mode: "0644"
       loop:
-        - { regexp: '^LANG=', line: 'LANG={{ locale }}' }
-        - { regexp: '^LC_ALL=', line: 'LC_ALL={{ locale }}' }
+        - { regexp: ^LANG=, line: "LANG={{ locale }}" }
+        - { regexp: ^LC_ALL=, line: "LC_ALL={{ locale }}" }
```
Comment (vitabaks) [inline roles/locales/tasks/main.yml:69]: > regexp: ^LANG=,

Are you sure this code is working now? Since you removed the quotes

Comment (vitabaks) [reply roles/locales/tasks/main.yml:69]: > "LANG={{ locale }}" 

why are double quotes better here than using single quotes?

File: vars/Debian.yml [hunk 9]
```diff
@@ -130,64 +129,52 @@ patroni_pip_package_repo: []
...
 packages_from_file: []
-#  - "my-package-name_1_amd64.deb"
-#  - "my-package-name_2_amd64.deb"
-#  - ""
-
-# ----------------------------------------------------------------------------------------------------------------------------------
-# Attention! If you want to use the installation method "file".
-# You need to independently determine all the necessary the dependencies of deb packages for your version of the Linux distribution.
-# ----------------------------------------------------------------------------------------------------------------------------------
-
```
Comment (vitabaks) [inline vars/Debian.yml:192]: why did you delete this warning?

**Comment by vitabaks** [comment]
I think we need to start by defining the style by code and describe it or provide links in the document CONTRIBUTING.md

Example:

https://lean-delivery.com/ansible-development-kit/docs/ansible_style_guide

**Comment by ThomasSanson** [comment]
@vitabaks, you're absolutely right, this pull request was not ideal.

To explain what happened: I removed the "yaml" section from the skip_list in ansible-lint and used the "--write" option in ansible-lint to automatically fix the files using this command: `ansible-lint --force-color --parseable --write`.

However, this resulted in too many file changes at once. I'm going to cancel this pull request and redo it, initially breaking down the skip_list part for yaml. This will allow us to progress more incrementally, which will be more effective.

I apologize for the time and energy spent on this task.

**Comment by vitabaks** [comment]
@ThomasSanson Ok. Nevertheless, I think it's a good idea to coordinate and describe the code style in CONTRIBUTING.md so that there are fewer questions when reviewing the code.
